### PR TITLE
Add the initial configuration of wazuh_table and pf to the .conf file…

### PR DESCRIPTION
**Pull request by thejbte**
|Related issues|Manual Testing|
|---|---|
|https://github.com/wazuh/wazuh/issues/14073|https://github.com/wazuh/wazuh-qa/issues/3330|

## Description
This PR solves the issue https://github.com/wazuh/wazuh/issues/14073, the proposed solution was that when a **wazuh_fwtable** does not exist, it is created , the configurations are added to the `pf.conf` files and `pf` rules are reload. Also  added an alert when `pf` services are not accessible, also added an alert when the firewall is disabled, it is verified if the firewall is down and if so, it is written in the active response log in order to be able to generate the alert, for this the rule (658) is used.

## Configuration options
Agent ossec.conf :
```xml
<ossec_config>
  <localfile>
    <location>active-response\active-responses.log</location>
    <log_format>syslog</log_format>
  </localfile>
</ossec_config>
```
### Configuration option for test
Agent ossec.conf :
```xml
<ossec_config>
  <localfile>
    <log_format>apache</log_format>
    <location>/var/log/apache2/access_log</location>
  </localfile>
</ossec_config>
```
Manager ossec.conf:
```xml
<ossec_config>
   <active-response>
     <command>firewall-drop</command>
     <location>local</location>
     <rules_id>100100</rules_id>
     <timeout>60</timeout>
   </active-response>
 </ossec_config>
```

```xml
<ossec_config>
      <ruleset>
          <!-- Default ruleset -->
          <decoder_dir>ruleset/decoders</decoder_dir>
          <rule_dir>ruleset/rules</rule_dir>
          <rule_exclude>0215-policy_rules.xml</rule_exclude>
          <list>etc/lists/audit-keys</list>
         <list>etc/lists/blacklist-alienvault</list>
          <!-- User-defined ruleset -->
          <decoder_dir>etc/decoders</decoder_dir>
          <rule_dir>etc/rules</rule_dir>
      </ruleset>

      <active-response>
          <command>firewall-drop</command>
          <location>local</location>
          <rules_id>100100</rules_id>
          <timeout>60</timeout>
      </active-response>
</ossec_config>
```

Rules:
``` xml 
<rule id="658" level="5">
  <if_sid>650</if_sid>
    <status>inactive</status>
    <description>Active response related to $(script) has been activated, but may not have an effect because the firewall is $(status)</description>
    <group>active_response,pci_dss_11.4,gdpr_IV_35.7.d,nist_800_53_SI.4,tsc_CC6.1,tsc_CC6.8,tsc_CC7.2,tsc_CC7.3,tsc_CC7.4,</group>
</rule>
```
### Option test
1. Set configuration option for test
2. Configure Blocking a malicious actor PoC [(Link)](https://documentation.wazuh.com/current/proof-of-concept-guide/block-malicious-actor-ip-reputation.html)
3. Generate the alert
4. Try to access the agent from ssh when it has generated an alert blocking a malicious actor, so that it verifies that the firewall has its IP blocked
5. Disable firewall `pfctl -d` , reinstall agent and repeat test again.
6. Enable firewall  `pfctl -e` and repeat test again.

<details><summary><b>Vagrantfile Mac OS (optional)</summary>
<p>

#### Steps:
1. vagrant up
2. if has error, disable usb  3.0 and set to 1.0
3. VBoxManage modifyvm "mojave_instance_bck" --cpu-profile "Intel Core i7-6700K"
4. vagrant up

```ruby
Vagrant.configure("2") do |config|
  config.vm.box = "danimaetrix/macOS-mojave"
  config.vm.box_version = "1.0"
  config.vm.network "private_network", ip: "192.168.56.80"
   config.vm.provider "virtualbox" do |vb|
     # Display the VirtualBox GUI when booting the machine
     vb.gui = true
     vb.name = "mojave_instance_bck"
     # Customize the amount of memory on the VM:
  #   vb.memory = "1024"
   end
 end
```
</p>
</details>

<details><summary><b>Vagrantfile freeBSD (optional)</summary>
<p>

#### Steps:
1. vagrant up
2. if has error, disable usb  3.0 and set to 1.0
3. vagrant up

```ruby
Vagrant.configure("2") do |config|
  config.vm.box = "bento/freebsd-13"
  config.vm.network "private_network", ip: "192.168.56.85"
end
```
</p>
</details>

<details><summary><b>Vagrantfile openBSD (optional)</summary>
<p>

#### Steps:
1. vagrant up
2. if has error, disable usb  3.0 and set to 1.0
3. vagrant up

```ruby
Vagrant.configure("2") do |config|
  config.vm.box = "twingly/openbsd-6.9-amd64"
  config.vm.box_version = "1.0.0"
  config.vm.network "private_network", ip: "192.168.56.86"
end
```
</p>
</details>
#### Alert when firewall is disabled
```vim
{"timestamp":"2022-09-22T14:45:11.365-0500","rule":{"level":5,"description":"Active response related to pf has been activated, but may not have an effect because the firewall is inactive","id":"658","firedtimes":2,"mail":false,"groups":["ossec","active_response"],"pci_dss":["11.4"],"gdpr":["IV_35.7.d"],"nist_800_53":["SI.4"],"tsc":["CC6.1","CC6.8","CC7.2","CC7.3","CC7.4"]},"agent":{"id":"009","name":"Vagrants-MacBook-Pro-backup1.local","ip":"FE80:0000:0000:0000:003F:0A8E:91CB:9874"},"manager":{"name":"thejbte-ASUS-TUF-A15"},"id":"1663875911.1808273","full_log":"2022/09/22 14:45:11 active-response/bin/firewall-drop: {\"message\":\"Active response may not have an effect\",\"profile\":\"default\",\"status\":\"inactive\",\"script\":\"pf\"}","decoder":{"parent":"ar_log_json","name":"ar_log_json"},"data":{"status":"inactive","message":"Active response may not have an effect","profile":"default","script":"pf"},"location":"/Library/Ossec/logs/active-responses.log"}
```
#### Alert when blocking a malicious actor 
```vim
{"timestamp":"2022-09-17T14:20:08.552-0500","rule":{"level":10,"description":"IP address found in AlienVault reputation database.","id":"100100","firedtimes":4,"mail":false,"groups":["attack"]},"agent":{"id":"006","name":"Vagrants-MacBook-Pro-Bck.local","ip":"FE80:0000:0000:0000:1876:67B9:B5D4:0765"},"manager":{"name":"thejbte-ASUS-TUF-A15"},"id":"1663442408.1315007","full_log":"192.168.56.66 - - [17/Sep/2022:14:20:06 -0500] \"GET / HTTP/1.1\" 304 -","decoder":{"name":"web-accesslog"},"data":{"protocol":"GET","srcip":"192.168.56.66","id":"304","url":"/"},"location":"/var/log/apache2/access_log"}


{"timestamp":"2022-09-22T14:50:59.873-0500","rule":{"level":3,"description":"Host Blocked by firewall-drop Active Response","id":"651","firedtimes":8,"mail":false,"groups":["ossec","active_response"],"pci_dss":["11.4"],"gpg13":["4.13"],"gdpr":["IV_35.7.d"],"nist_800_53":["SI.4"],"tsc":["CC6.1","CC6.8","CC7.2","CC7.3","CC7.4"]},"agent":{"id":"009","name":"Vagrants-MacBook-Pro-backup1.local","ip":"FE80:0000:0000:0000:003F:0A8E:91CB:9874"},"manager":{"name":"thejbte-ASUS-TUF-A15"},"id":"1663876259.1837832","full_log":"2022/09/22 14:50:57 active-response/bin/firewall-drop: {\"version\":1,\"origin\":{\"name\":\"node01\",\"module\":\"wazuh-execd\"},\"command\":\"add\",\"parameters\":{\"extra_args\":[],\"alert\":{\"timestamp\":\"2022-09-22T14:50:57.854-0500\",\"rule\":{\"level\":10,\"description\":\"IP address found in AlienVault reputation database.\",\"id\":\"100100\",\"firedtimes\":8,\"mail\":false,\"groups\":[\"attack\"]},\"agent\":{\"id\":\"009\",\"name\":\"Vagrants-MacBook-Pro-backup1.local\",\"ip\":\"FE80:0000:0000:0000:003F:0A8E:91CB:9874\"},\"manager\":{\"name\":\"thejbte-ASUS-TUF-A15\"},\"id\":\"1663876257.1837528\",\"full_log\":\"192.168.56.66 - - [22/Sep/2022:14:50:57 -0500] \\\"GET / HTTP/1.1\\\" 304 -\",\"decoder\":{\"name\":\"web-accesslog\"},\"data\":{\"protocol\":\"GET\",\"srcip\":\"192.168.56.66\",\"id\":\"304\",\"url\":\"/\"},\"location\":\"/var/log/apache2/access_log\"},\"program\":\"active-response/bin/firewall-drop\"}}","decoder":{"parent":"ar_log_json","name":"ar_log_json"},"data":{"version":"1","origin":{"name":"node01","module":"wazuh-execd"},"command":"add","parameters":{"extra_args":[],"alert":{"timestamp":"2022-09-22T14:50:57.854-0500","rule":{"level":"10","description":"IP address found in AlienVault reputation database.","id":"100100","firedtimes":"8","mail":"false","groups":["attack"]},"agent":{"id":"009","name":"Vagrants-MacBook-Pro-backup1.local","ip":"FE80:0000:0000:0000:003F:0A8E:91CB:9874"},"manager":{"name":"thejbte-ASUS-TUF-A15"},"id":"1663876257.1837528","full_log":"192.168.56.66 - - [22/Sep/2022:14:50:57 -0500] \"GET / HTTP/1.1\" 304 -","decoder":{"name":"web-accesslog"},"data":{"protocol":"GET","srcip":"192.168.56.66","id":"304","url":"/"},"location":"/var/log/apache2/access_log"},"program":"active-response/bin/firewall-drop"}},"location":"/Library/Ossec/logs/active-responses.log"}
```
#### Alert when unblocking a malicious actor 
```vim
{"timestamp":"2022-09-22T14:52:00.032-0500","rule":{"level":3,"description":"Host Unblocked by firewall-drop Active Response","id":"652","firedtimes":7,"mail":false,"groups":["ossec","active_response"],"pci_dss":["11.4"],"gpg13":["4.13"],"gdpr":["IV_35.7.d"],"nist_800_53":["SI.4"],"tsc":["CC6.1","CC6.8","CC7.2","CC7.3","CC7.4"]},"agent":{"id":"009","name":"Vagrants-MacBook-Pro-backup1.local","ip":"FE80:0000:0000:0000:003F:0A8E:91CB:9874"},"manager":{"name":"thejbte-ASUS-TUF-A15"},"id":"1663876320.1840055","full_log":"2022/09/22 14:51:58 active-response/bin/firewall-drop: {\"version\":1,\"origin\":{\"name\":\"node01\",\"module\":\"wazuh-execd\"},\"command\":\"delete\",\"parameters\":{\"extra_args\":[],\"alert\":{\"timestamp\":\"2022-09-22T14:50:57.854-0500\",\"rule\":{\"level\":10,\"description\":\"IP address found in AlienVault reputation database.\",\"id\":\"100100\",\"firedtimes\":8,\"mail\":false,\"groups\":[\"attack\"]},\"agent\":{\"id\":\"009\",\"name\":\"Vagrants-MacBook-Pro-backup1.local\",\"ip\":\"FE80:0000:0000:0000:003F:0A8E:91CB:9874\"},\"manager\":{\"name\":\"thejbte-ASUS-TUF-A15\"},\"id\":\"1663876257.1837528\",\"full_log\":\"192.168.56.66 - - [22/Sep/2022:14:50:57 -0500] \\\"GET / HTTP/1.1\\\" 304 -\",\"decoder\":{\"name\":\"web-accesslog\"},\"data\":{\"protocol\":\"GET\",\"srcip\":\"192.168.56.66\",\"id\":\"304\",\"url\":\"/\"},\"location\":\"/var/log/apache2/access_log\"},\"program\":\"active-response/bin/firewall-drop\"}}","decoder":{"parent":"ar_log_json","name":"ar_log_json"},"data":{"version":"1","origin":{"name":"node01","module":"wazuh-execd"},"command":"delete","parameters":{"extra_args":[],"alert":{"timestamp":"2022-09-22T14:50:57.854-0500","rule":{"level":"10","description":"IP address found in AlienVault reputation database.","id":"100100","firedtimes":"8","mail":"false","groups":["attack"]},"agent":{"id":"009","name":"Vagrants-MacBook-Pro-backup1.local","ip":"FE80:0000:0000:0000:003F:0A8E:91CB:9874"},"manager":{"name":"thejbte-ASUS-TUF-A15"},"id":"1663876257.1837528","full_log":"192.168.56.66 - - [22/Sep/2022:14:50:57 -0500] \"GET / HTTP/1.1\" 304 -","decoder":{"name":"web-accesslog"},"data":{"protocol":"GET","srcip":"192.168.56.66","id":"304","url":"/"},"location":"/var/log/apache2/access_log"},"program":"active-response/bin/firewall-drop"}},"location":"/Library/Ossec/logs/active-responses.log"}
```

## Tests

- Compilation without warnings in every supported platform
  - [x] MAC OS X
- [x] Source installation
- [x] Package installation
- [x] Source upgrade
- [x] Package upgrade
- [x] Review logs syntax and correct language

### CODE REVIEW

File: src/active-response/firewalls/pf.c [hunk 1]
```diff
@@ -164,3 +270,45 @@ static int checking_if_its_configured(const char *path, const char *table) {
...
+    if (path != NULL && cmd != NULL) {
+        FILE *fp = fopen(path, "a+");
+        if (fp != NULL) {
+    	    fprintf(fp, "%s\n", cmd);
+	        retVal = 1;
+        }
+        fclose(fp);
+    }
+    return retVal;
+ }
```
Comment (TomasTurina) [inline src/active-response/firewalls/pf.c:314]: Always add a new line at the end of the file.

File: src/active-response/firewalls/pf.c [hunk 2]
```diff
@@ -164,3 +270,45 @@ static int checking_if_its_configured(const char *path, const char *table) {
...
+    }
+    return  retVal;
+}
+
+static int write_cmd_to_file(const char *path, const char *cmd) {
+    int retVal = 0;
+    if (path != NULL && cmd != NULL) {
+        FILE *fp = fopen(path, "a+");
+        if (fp != NULL) {
+    	    fprintf(fp, "%s\n", cmd);
```
Comment (TomasTurina) [inline src/active-response/firewalls/pf.c:308]:
```suggestion
            fprintf(fp, "%s\n", cmd);
```

File: src/active-response/firewalls/pf.c [hunk 3]
```diff
@@ -164,3 +270,45 @@ static int checking_if_its_configured(const char *path, const char *table) {
...
+	        retVal = 1;
```
Comment (TomasTurina) [inline src/active-response/firewalls/pf.c:309]:
```suggestion
            retVal = 1;
```

File: src/active-response/firewalls/pf.c [hunk 4]
```diff
@@ -164,3 +270,45 @@ static int checking_if_its_configured(const char *path, const char *table) {
...
+
+/*
+   retVal = isEnabledFromPattern(output_buf, "Status: ", "Enabled");
+   return 1 if for example with patter1:"Status :" find "Enabled"
+   
+*/
+static int isEnabledFromPattern(const char * output_buf, const char * str_pattern_1, const char * str_pattern_2){
+    int retVal = 0;
+    const char *pos = NULL;
+    if ( str_pattern_1 != NULL) {
```
Comment (TomasTurina) [inline src/active-response/firewalls/pf.c:282]:
```suggestion
    if (str_pattern_1 != NULL) {
```

File: src/active-response/firewalls/pf.c [hunk 5]
```diff
@@ -164,3 +270,45 @@ static int checking_if_its_configured(const char *path, const char *table) {
...
+        pos = strstr(output_buf, str_pattern_1);
+    }
+    
+    if ( pos != NULL) {
```
Comment (TomasTurina) [inline src/active-response/firewalls/pf.c:286]:
```suggestion
    if (pos != NULL) {
```

File: src/active-response/firewalls/pf.c [hunk 6]
```diff
@@ -164,3 +270,45 @@ static int checking_if_its_configured(const char *path, const char *table) {
...
+        char state[OS_MAXSTR];
+        char buffer[OS_MAXSTR];
+        if (str_pattern_2 != NULL) {
+            snprintf(buffer, OS_MAXSTR -1 , "%%*s %%%lds",strlen(str_pattern_2));
+            if (pos && sscanf(pos, buffer /*"%*s %7s"*/, state) == 1 ) {
```
Comment (TomasTurina) [inline src/active-response/firewalls/pf.c:291]:
```suggestion
            if (pos && sscanf(pos, buffer /*"%*s %7s"*/, state) == 1) {
```

File: src/active-response/firewalls/pf.c [hunk 7]
```diff
@@ -164,3 +270,45 @@ static int checking_if_its_configured(const char *path, const char *table) {
...
+    const char *pos = NULL;
+    if ( str_pattern_1 != NULL) {
+        pos = strstr(output_buf, str_pattern_1);
+    }
+    
+    if ( pos != NULL) {
+        char state[OS_MAXSTR];
+        char buffer[OS_MAXSTR];
+        if (str_pattern_2 != NULL) {
+            snprintf(buffer, OS_MAXSTR -1 , "%%*s %%%lds",strlen(str_pattern_2));
```
Comment (TomasTurina) [inline src/active-response/firewalls/pf.c:290]:
```suggestion
            snprintf(buffer, OS_MAXSTR -1 , "%%*s %%%lds", strlen(str_pattern_2));
```

File: src/active-response/firewalls/pf.c [hunk 8]
```diff
@@ -164,3 +270,45 @@ static int checking_if_its_configured(const char *path, const char *table) {
     }
     return OS_INVALID;
 }
+
+/*
+   retVal = isEnabledFromPattern(output_buf, "Status: ", "Enabled");
+   return 1 if for example with patter1:"Status :" find "Enabled"
+
```
Comment (TomasTurina) [inline src/active-response/firewalls/pf.c:277]: Remove blank space

File: src/active-response/firewalls/pf.c [hunk 9]
```diff
@@ -101,6 +121,48 @@ int main (int argc, char **argv) {
                 snprintf(log_msg, OS_MAXSTR - 1, "Table '%s' does not exist", PFCTL_TABLE);
                 write_debug_file(argv[0], log_msg);
                 cJSON_Delete(input_json);
+                
+                int retVal = 0;
```
Comment (TomasTurina) [inline src/active-response/firewalls/pf.c:125]: You are not using this result anywhere.

File: src/active-response/firewalls/pf.c [hunk 10]
```diff
@@ -101,6 +121,48 @@ int main (int argc, char **argv) {
...
+                
+                memset(log_msg, '\0', OS_MAXSTR);
+                snprintf(log_msg, OS_MAXSTR - 1, "block in quick from <%s> to any", PFCTL_TABLE);
+                retVal = write_cmd_to_file(PFCTL_RULES, log_msg);
+                
+                memset(log_msg, '\0', OS_MAXSTR);
+                snprintf(log_msg, OS_MAXSTR - 1, "block out quick from any to <%s>", PFCTL_TABLE);
+                retVal = write_cmd_to_file(PFCTL_RULES, log_msg);
+
+                if (exec_cmd4[0] && strcmp(exec_cmd4[0], PFCTL) == 0) {
```
Comment (TomasTurina) [inline src/active-response/firewalls/pf.c:143]: When will this code executed? At this point, `exec_cmd4` will always be NULL.

File: src/active-response/firewalls/pf.c [hunk 11]
```diff
@@ -101,6 +121,48 @@ int main (int argc, char **argv) {
...
+                        memset(log_msg, '\0', OS_MAXSTR);
+                        snprintf(log_msg, OS_MAXSTR - 1, "Error executing '%s' : %s", PFCTL, strerror(errno));
+                        write_debug_file(argv[0], log_msg);
+                        cJSON_Delete(input_json);
+                        return OS_INVALID;
+                    }
+                    wpclose(wfd);
+                }
+
+                if (exec_cmd5[0] && strcmp(exec_cmd5[0], PFCTL) == 0) {
```
Comment (TomasTurina) [inline src/active-response/firewalls/pf.c:155]: Same here.

File: src/active-response/firewalls/pf.c [hunk 12]
```diff
@@ -112,6 +174,50 @@ int main (int argc, char **argv) {
...
+                            "undefined", 1 == isEnabledFirewall ? "active" : "inactive" 
+                    );
+                    write_debug_file(argv[0], log_msg);
+                }
+            }
+            wpclose(wfd);
+
+
+            // reload pf.conf after detected that user turn on firewall
+            if ( 0 == isFirstEntryEnabledFirewall || (0 == isEnabledFirewallPrevious && 1 == isEnabledFirewall)){
```
Comment (TomasTurina) [inline src/active-response/firewalls/pf.c:204]: I think these static variables won't persist between script executions.

**Review by TomasTurina — CHANGES_REQUESTED** [review]

File: src/active-response/firewalls/pf.c [hunk 13]
```diff
@@ -101,6 +110,35 @@ int main (int argc, char **argv) {
                 snprintf(log_msg, OS_MAXSTR - 1, "Table '%s' does not exist", PFCTL_TABLE);
                 write_debug_file(argv[0], log_msg);
                 cJSON_Delete(input_json);
+
+                (void)write_cmd_to_file(RCCONF, "pf_enable=\"YES\"" "\n" "pf_rules=\"/etc/pf.conf\"" "\n" "pflog_enable=\"YES\"" "\n" "pflog_logfile=\"/var/log/pflog\"");
```
Comment (TomasTurina) [inline src/active-response/firewalls/pf.c:114]: Why don't you verify the return value?

File: src/active-response/firewalls/pf.c [hunk 14]
```diff
@@ -101,6 +110,35 @@ int main (int argc, char **argv) {
...
+                memset(log_msg, '\0', OS_MAXSTR);
+                snprintf(log_msg, OS_MAXSTR - 1, "table <%s> persist #%s\nblock in quick from <%s> to any\nblock out quick from any to <%s>", PFCTL_TABLE, PFCTL_TABLE, PFCTL_TABLE, PFCTL_TABLE);
+                (void)write_cmd_to_file(PFCTL_RULES, log_msg);
```
Comment (TomasTurina) [inline src/active-response/firewalls/pf.c:117]: Why don't you verify the return value?

File: src/active-response/firewalls/pf.c (hunk 13 above)
Comment (TomasTurina) [reply src/active-response/firewalls/pf.c:114]: `"pf_enable=\"YES\"\npf_rules=\"/etc/pf.conf\"\npflog_enable=\"YES\"\npflog_logfile=\"/var/log/pflog\""`

File: src/active-response/firewalls/pf.c [hunk 15]
```diff
@@ -101,6 +110,35 @@ int main (int argc, char **argv) {
...
+
+                if (exec_cmd4[0] != NULL ) {
+                    wfd = wpopenv(PFCTL, exec_cmd4, W_BIND_STDOUT);
```
Comment (TomasTurina) [inline src/active-response/firewalls/pf.c:120]: Maybe `LAUNCHCTL`? Will this work in FreeBSD and OpenBSD?

File: src/active-response/firewalls/pf.c [hunk 16]
```diff
@@ -101,6 +110,35 @@ int main (int argc, char **argv) {
...
+                    if (!wfd) {
+                        memset(log_msg, '\0', OS_MAXSTR);
+                        snprintf(log_msg, OS_MAXSTR - 1, "Error executing '%s' : %s", PFCTL, strerror(errno));
+                        write_debug_file(argv[0], log_msg);
+                        cJSON_Delete(input_json);
+                        return OS_INVALID;
+                    }
+                    wpclose(wfd);
+                }
                 return OS_SUCCESS;
```
Comment (TomasTurina) [inline src/active-response/firewalls/pf.c:142]: Table is created but AR is not executed.

File: src/active-response/firewalls/pf.c [hunk 17]
```diff
@@ -112,7 +150,33 @@ int main (int argc, char **argv) {
...
+                write_debug_file(argv[0], log_msg);
+                cJSON_Delete(input_json);
+                return OS_INVALID;
+            }
+            else {
+                while (fgets(output_buf, OS_MAXSTR -1, wfd->file_out) && 0  == isEnabledFirewall) {
+                    isEnabledFirewall = isEnabledFromPattern(output_buf, "Status: ", "Enabled");
+                }
+
+                if (( 0 == isEnabledFirewall)) {
```
Comment (TomasTurina) [inline src/active-response/firewalls/pf.c:168]:
```suggestion
                if (0 == isEnabledFirewall) {
```

File: src/active-response/firewalls/pf.c [hunk 18]
```diff
@@ -164,3 +228,55 @@ static int checking_if_its_configured(const char *path, const char *table) {
...
     return OS_INVALID;
 }
+
+/**
+ * @brief search a pair pattern1 and after pattern2
+ * @param output_buf buffer where search
+ * @param str_pattern_1 pattern search to match
+ * @param str_pattern_2 pattern search to match
+ * @return 1 or 0 
+ * @example retVal = isEnabledFromPattern(output_buf, "Status: ", "Enabled");
```
Comment (TomasTurina) [inline src/active-response/firewalls/pf.c:238]: This example is confusing and English should be improved.

File: src/active-response/firewalls/pf.c [hunk 19]
```diff
@@ -164,3 +228,55 @@ static int checking_if_its_configured(const char *path, const char *table) {
...
+    const char *pos = NULL;
+    if (str_pattern_1 != NULL) {
+        pos = strstr(output_buf, str_pattern_1);
+    }
+
+    if (pos != NULL) {
+        char state[OS_MAXSTR];
+        char buffer[OS_MAXSTR];
+        if (str_pattern_2 != NULL) {
+            snprintf(buffer, OS_MAXSTR -1 , "%%*s %%%lds",strlen(str_pattern_2));
```
Comment (TomasTurina) [inline src/active-response/firewalls/pf.c:252]:
```suggestion
            snprintf(buffer, OS_MAXSTR -1 , "%%*s %%%lds", strlen(str_pattern_2));
```

File: src/active-response/firewalls/pf.c [hunk 20]
```diff
@@ -164,3 +228,55 @@ static int checking_if_its_configured(const char *path, const char *table) {
...
+            if (pos && sscanf(pos, buffer /*"%*s %7s"*/, state) == 1) {
```
Comment (TomasTurina) [inline src/active-response/firewalls/pf.c:253]: Why do you check `pos` not null again?

**Review by TomasTurina — CHANGES_REQUESTED** [review]

**Review by TomasTurina — APPROVED** [review]

**Review by TomasTurina — DISMISSED** [review]

**Review by TomasTurina — APPROVED** [review]

**Review by TomasTurina — APPROVED** [review]

**Comment by damarisg** [comment]
## QA review
- **Type**: Manual testing.
- **Status**: Approved  🟢 
- **Testing issue**: https://github.com/wazuh/wazuh-qa/issues/3330
- **Comments**:  https://github.com/wazuh/wazuh-qa/issues/3330#issuecomment-1450333730

**Pull request merged**